src/pages/сustom.tsx

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useContext, useState, useEffect } from "react";
+import {
+    Card,
+    CardContent,
+    CardMedia,
+    Typography,
+    Box,
+    IconButton,
+    Rating,
+    Grid,
+    Accordion,
+    AccordionSummary,
+    AccordionDetails,
+    FormControl,
+    InputLabel,
+    Select,
+    MenuItem,
+    TextField,
+    RadioGroup,
+    FormControlLabel,
+    Radio,
+    Divider
+} from "@mui/material";
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import { styled } from '@mui/system';
+import { useSearchParams } from 'react-router-dom';
+import { MotoContext } from "../components/providers/motoProvider";
+import { CarsContext } from "../components/providers/carsProvider";
+import { LocaleContext } from "../components/providers/localeProvider";
+interface VehicleCardProps {
+    vehicle: any;
+    engineTypes: string[];
+    toggleFavorite: (id: number) => void;
+}
+
+const VehicleCard: React.FC<VehicleCardProps> = ({ vehicle, engineTypes, toggleFavorite }) => {
+    return (
+        <Card>
+            <CardMedia
+                component="img"
+                alt={vehicle.model}
+                height="140"
+                image={vehicle.image}
+                title={vehicle.model}
+            />
+            <CardContent>
+                <Typography variant="h5" component="div">
+                    {vehicle.model}
+                </Typography>
+                <Typography variant="body1" color="text.secondary">
+                    {vehicle.maker}, {vehicle.year}
+                </Typography>
+                <Typography variant="body2" color="text.secondary">
+                    {vehicle.engine !== undefined && engineTypes[vehicle.engine]}
+                </Typography>
+                <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between', mt: 1 }}>
+                    <Rating value={3} readOnly />
+                    <IconButton color='error' onClick={() => toggleFavorite(vehicle.id)}>
+                        {vehicle.favorite ? <FavoriteIcon /> : <FavoriteBorderIcon />}
+                    </IconButton>
+                </Box>
+            </CardContent>
+        </Card>
+    );
+};
+
+const Sidebar = styled(Box)(({ theme }) => ({
+    width: '20%',
+    padding: theme.spacing(2),
+    borderRight: `1px solid ${theme.palette.divider}`,
+    [theme.breakpoints.down('md')]: {
+        width: '100%',
+        borderRight: 'none',
+        borderBottom: `1px solid ${theme.palette.divider}`
+    }
+}));
```

Comment:
Try to always split components. For example here, side bar can be separate component.
This way it will significantly reduce line of codes in root page file